File: CzhERP.Application/Entity/PurOrderItem.cs
namespace CzhERP.Application.Entity;

[SugarTable("Pur_OrderItem", TableDescription = "采购订单明细表")]
public class PurOrderItem : EntityBaseId
{
    [SugarColumn(ColumnDescription = "订单ID", IsNullable = false)]
    public long OrderId { get; set; }

    [SugarColumn(ColumnDescription = "物料ID", IsNullable = false)]
    public long MaterialId { get; set; }

    [SugarColumn(ColumnDescription = "物料编码", Length = 50, IsNullable = false)]
    public string MaterialCode { get; set; }

    [SugarColumn(ColumnDescription = "物料名称", Length = 100, IsNullable = false)]
    public string MaterialName { get; set; }

    [SugarColumn(ColumnDescription = "规格型号", Length = 100)]
    public string? Spec { get; set; }

    [SugarColumn(ColumnDescription = "单位ID")]
    public long? UnitId { get; set; }

    [SugarColumn(ColumnDescription = "单位名称", Length = 50, IsNullable = false)]
    public string UnitName { get; set; }

    [SugarColumn(ColumnDescription = "订单数量", IsNullable = false)]
    public decimal OrderQty { get; set; }

    [SugarColumn(ColumnDescription = "单价(不含税)", IsNullable = false)]
    public decimal Price { get; set; }

    [SugarColumn(ColumnDescription = "金额(不含税)", DefaultValue = "0")]
    public decimal Amount { get; set; } = 0;

    [SugarColumn(ColumnDescription = "税率", DefaultValue = "0")]
    public decimal TaxRate { get; set; } = 0;

    [SugarColumn(ColumnDescription = "税额", DefaultValue = "0")]
    public decimal TaxAmount { get; set; } = 0;

    [SugarColumn(ColumnDescription = "交货日期")]
    public DateTime? DeliveryDate { get; set; }

    [SugarColumn(ColumnDescription = "已收货数量", DefaultValue = "0")]
    public decimal ReceivedQty { get; set; } = 0;

    [SugarColumn(ColumnDescription = "备注", Length = 200)]
    public string? Remark { get; set; }

    [SugarColumn(ColumnDescription = "排序", DefaultValue = "0")]
    public int SortOrder { get; set; } = 0;

    [Navigate(NavigateType.OneToOne, nameof(OrderId))]
    public PurOrder? Order { get; set; }
}

File: Web/src/api/purchase/purOrderItem.ts
import {useBaseApi} from '/@/api/base';

// 采购订单明细表接口服务
export const usePurOrderItemApi = () => {
	const baseApi = useBaseApi("purOrderItem");
	return {
		// 分页查询采购订单明细表
		page: baseApi.page,
		// 查看采购订单明细表详细
		detail: baseApi.detail,
		// 新增采购订单明细表
		add: baseApi.add,
		// 更新采购订单明细表
		update: baseApi.update,
		// 删除采购订单明细表
		delete: baseApi.delete,
		// 批量删除采购订单明细表
		batchDelete: baseApi.batchDelete,
		// 导出采购订单明细表数据
		exportData: baseApi.exportData,
		// 导入采购订单明细表数据
		importData: baseApi.importData,
		// 下载采购订单明细表数据导入模板
		downloadTemplate: baseApi.downloadTemplate,
	}
}

// 采购订单明细表实体
export interface PurOrderItem {
	// 主键Id
	id: number;
	// 订单ID
	orderId?: number;
	// 物料ID
	materialId?: number;
	// 物料编码
	materialCode?: string;
	// 物料名称
	materialName?: string;
	// 规格型号
	spec: string;
	// 单位ID
	unitId: number;
	// 单位名称
	unitName?: string;
	// 订单数量
	orderQty?: number;
	// 单价(不含税)
	price?: number;
	// 金额(不含税)
	amount?: number;
	// 税率
	taxRate?: number;
	// 税额
	taxAmount?: number;
	// 交货日期
	deliveryDate: string;
	// 已收货数量
	receivedQty?: number;
	// 备注
	remark: string;
	// 排序
	sortOrder?: number;
}